ompi/mpi/java/c/mpi_MPI.c

```diff
@@ -1154,6 +1153,9 @@ jboolean ompi_java_exceptionCheck(JNIEnv *env, int rc)
         (*env)->DeleteLocalRef(env, jmessage);
         return JNI_TRUE;
     }
+    else if (JNI_TRUE == jni_exception) {
+        return JNI_TRUE;
```

Comment:
@ggouaillardet @hppritcha @jsquyres @siegmargross 

I would say no.  JNI exceptions are directly related to problems with something the JNI has tried to do.  It seems to me like this is a separate issue from MPI because it is a result of a user trying to do something they shouldn't in Java.  In the case of the index out of bounds error that has brought this up, a java array is too short for the attempted call.

That said, I have thoroughly traced the way the exceptions are being thrown, and discovered that it is actually acting differently from the way I had originally thought.  The ompi_java_exceptionCheck method is returning a true when a JNI error occurs, however it's return is almost never checked (you can do a "git grep ompi_java_exceptionCheck" from the top ompi directory to see what I am talking about).

Instead, in almost every case, the code continues to execute in the C code, and the Java exception is not actually thrown until it returns to Java land.

I propose we change it to either throw an exception in ompi_java_exceptionCheck back to Java so the code stops executing and change the method return to void, or change all of the places where ompi_java_exceptionCheck is being called to actually check the return value.

The first option is far less code, but the second option would allow us to do memory cleanup stuff.  Any thoughts?